0002-gitops-management-clusters/0002-gitops-management-clusters.md

```diff
@@ -0,0 +1,27 @@
+# RFC 0002 - Enable customers to use gitops in management clusters
+
+We currently give customers access to the management clusters but do not support them in utilizing this access effectively in terms of git ops related management (e.g. for apps).
+
+Context:
+Some customers might be interested in utilizing mainstream git ops tooling in order to manage their apps across clusters and installations.
+
+## Open Questions
+
+### 1. General direction
+1.1 We would like to allow our customers to choose their management tooling freely. Do we want to support some tools with e.g. a managed app first though?
+
+1.2 Do we own the gitops tooling or is it purely owned by the customer?
+
+### 2. Technical issues
+2.1 How does the setup for an in-cluster agent look like? Currently customers will struggle setting up an in-cluster agent with appropriate permissions.
```

Comment:
In the app CR the customer just needs to set .spec.kubeConfig.inCluster to true. 

We should also make our kubeconfigs work with GitOps tooling. They don't currently work with Flux because of how we name the secrets we generate in cluster-operator.

Reply:
Sorry this was bad phrasing on my part:
This more refers to the initial agent set-up (at least as long as we don't have a managed app). We will currently struggle to not give the agent full permissions to all namespaces, right?

Reply:
> This more refers to the initial agent set-up (at least as long as we don't have a managed app). We will currently struggle to not give the agent full permissions to all namespaces, right?

Ah got it. Yes I think we'll need to restrict the permissions. Taking Flux as an example AIUI there are multi tenancy patterns with a system Flux for platform admins and then multiple Flux's for teams.

My proposal would be we follow that approach and offer a managed app with the locked down setup. This is another argument for offering managed apps for this IMO.